website/content/en/docs/cloud-providers/AWS/launch-templates.md

```diff
@@ -0,0 +1,233 @@
+
+Resources referenced by these fields are controlled by EKS/Karpenter, and not the launch template. 
+
+### Instance Type
+
+The instance type should not be specified in the launch template. Karpenter
+will determine the launch template at run time. 
+
+### Network Interfaces
+
+The [AWS CNI](https://docs.aws.amazon.com/eks/latest/userguide/pod-networking.html) will configure the network interfaces. Do not configure network instances in the launch template.
+
+## Creating the Launch Template
+
+Launch Templates may be created via the web console, the AWS CLI, or
+CloudFormation. 
+
+### CloudFormation
+
+The procedure, in summary, is to:
+1. [Create an AMI as described in the EC2 documentation.](https://docs.aws.amazon.com/vm-import/latest/userguide/vmimport-image-import.html)
+2. Write a EC2 Launch Template specification including the AMI.
+3. Push the specification to AWS with CloudFormation.
+4. Update the Provisioner CRD to specify the new Launch Template.
+
+An example yaml cloudformation definition of a launch template for Karpenter is
+provided below. 
+
+Cloudformation yaml is suited for the moderately high configuration density of
+launch templates, and creating the unusual InstanceProfile resource. 
+
+You must manually replace these values in the template:
+- SecurityGroupID
+  - list all security groups with `aws ec2 describe-security-groups`
+- Parameters in UserData
+- AMI
+
+```yaml
+AWSTemplateFormatVersion: '2010-09-09'
+Resources:
+  # create InstanceProfile wrapper on NodeRole
+  KarpenterNodeInstanceProfile:
+    Type: "AWS::IAM::InstanceProfile"
+    Properties:
+      InstanceProfileName: "KarpenterNodeInstanceProfile"
+      Path: "/"
+      Roles:
+        - Ref: "KarpenterNodeRole"
+  # create role with basic permissions for EKS node
+  KarpenterNodeRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      RoleName: "KarpenterNodeRole"
+      Path: /
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                !Sub "ec2.${AWS::URLSuffix}"
+            Action:
+              - "sts:AssumeRole"
+      ManagedPolicyArns:
+        - !Sub "arn:${AWS::Partition}:iam::aws:policy/AmazonEKSWorkerNodePolicy"
+        - !Sub "arn:${AWS::Partition}:iam::aws:policy/AmazonEKS_CNI_Policy"
+        - !Sub "arn:${AWS::Partition}:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly"
+        - !Sub "arn:${AWS::Partition}:iam::aws:policy/AmazonSSMManagedInstanceCore"
+  MyLaunchTemplate:
+    Type: AWS::EC2::LaunchTemplate
+    Properties: 
+      LaunchTemplateData: 
+        IamInstanceProfile:
+          # Get ARN of InstanceProfile defined above
+          Arn: !GetAtt
+            - KarpenterInstanceProfile
+            - Arn
+        ImageId: ami-074cce78125f09d61
+        # UserData is Base64 Encoded
+        UserData: !Base64 > 
+            #!/bin/bash
+            /etc/eks/bootstrap.sh <my-cluster-name> \

```

Comment:
If you're going to use actual values elsewhere in this example, why not also in the UserData?

Reply:
this is a good point. i'll update it to use actual values.